.github/workflows/pr-check.yml

```diff
@@ -71,3 +71,47 @@ jobs:
       - name: Require changeset to be present in PR
         if: github.event.pull_request.user.login != 'dependabot[bot]'
         run: pnpm changeset status --since origin/main
+
+  compiler-e2e:
```

Comment:
Why this is a separate job:
- this test should probably be optional for now
- it's easier to see which tests failed exactly, e2e or the usual ones

It needs the previous check to pass because if usual tests fail, I don't think it makes a lot of sense to run e2e.